include/sway/server.h

```diff
@@ -89,6 +90,8 @@ struct sway_server {
 	struct wl_listener output_manager_apply;
 	struct wl_listener output_manager_test;
 
+	struct wlr_session_lock_manager_v1 *session_lock;
```

Comment:
I'd prefer to store `sway_session_lock_manager` here (maybe the struct could even be embedded so that we don't need to deal with alloc errors). That way we don't need to go through a `void *data` pointer to get back the `sway_session_lock_manager`.